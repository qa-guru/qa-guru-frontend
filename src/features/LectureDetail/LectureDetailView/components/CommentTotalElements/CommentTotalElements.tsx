import React from "react";
import { Stack, Typography } from "@mui/material";
import { ICommentTotalElements } from "./CommentTotalElements.types";

const CommentTotalElements: React.FC<ICommentTotalElements> = ({
  totalElements,
}) => {
  return (
    <Stack spacing={1} direction="row">
      <Typography variant="h5">Комментарии</Typography>
      <Typography variant="h5">({totalElements})</Typography>
    </Stack>
  );
};

export default CommentTotalElements;
