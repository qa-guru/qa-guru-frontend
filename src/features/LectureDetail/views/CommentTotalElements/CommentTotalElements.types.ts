import { CommentsHomeWorkByHomeWorkQuery } from "../../../../api/graphql/generated/graphql";

export interface ICommentTotalElements {
  totalElements?: number;
}
