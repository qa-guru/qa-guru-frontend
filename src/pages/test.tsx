import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
