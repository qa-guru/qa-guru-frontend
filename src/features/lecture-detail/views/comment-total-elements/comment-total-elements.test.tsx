import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";
import CommentTotalElements from "./comment-total-elements";

describe("CommentTotalElements", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CommentTotalElements totalElements={7} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
