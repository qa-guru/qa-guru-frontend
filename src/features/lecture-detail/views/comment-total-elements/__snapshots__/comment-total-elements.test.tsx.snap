// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentTotalElements > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1ohoath-MuiStack-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-19fr98w-MuiTypography-root"
    >
      Комментарии
    </h5>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-19fr98w-MuiTypography-root"
    >
      (7)
    </h5>
  </div>
</DocumentFragment>
`;
